File: frontend/src/api/theme.ts
const BASE = "http://localhost:8000";

export interface ThemeExpansion {
  world_name: string;
  attribute_names: Record<string, string>;
  currency_name: string;
  biome_family_names: Record<string, string>;
}

// Read-only, non-persisting -- lets BasicInfoStep's custom-theme path pre-fill
// naming suggestions from a one-line pitch. Callers should treat failure as
// non-fatal and fall back to blank/default values (matches map preview's role).
export async function expandTheme(pitch: string): Promise<ThemeExpansion> {
  const res = await fetch(`${BASE}/theme/expand`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ pitch }),
  });
  if (!res.ok) throw new Error(await res.text());
  return res.json() as Promise<ThemeExpansion>;
}

File: backend/routers/theme.py
from pydantic import BaseModel, ValidationError
from fastapi import APIRouter, HTTPException
from ..ai_provider import get_provider
from ..utils.theme_prompts import build_theme_expand_prompt

router = APIRouter()


class ThemeExpandRequest(BaseModel):
    pitch: str


class ThemeExpandResponse(BaseModel):
    world_name: str
    attribute_names: dict[str, str]
    currency_name: str
    biome_family_names: dict[str, str]


@router.post("/theme/expand", response_model=ThemeExpandResponse)
async def expand_theme(payload: ThemeExpandRequest):
    """Read-only, non-persisting -- lets the wizard's custom-theme path pre-fill Basic
    Info / World Bible with AI-suggested naming, exactly like map preview pre-fills
    WorldGen. Nothing here is saved; the wizard just holds the returned bundle in
    local draft state until Launch."""
    provider = get_provider()
    prompt = build_theme_expand_prompt(payload.pitch)
    try:
        result = await provider.generate(prompt)
    except (ValueError, Exception) as exc:
        raise HTTPException(502, f"AI provider error during theme expansion: {exc}") from exc
    try:
        return ThemeExpandResponse(**result)
    except ValidationError as exc:
        raise HTTPException(500, f"DM did not return a valid theme bundle: {exc}") from exc
